ceph.yaml

```diff
@@ -0,0 +1,344 @@
+          --cap osd 'allow *' \
+          --cap mds 'allow *' \
+          --cap mgr 'allow *'
+
+        # Ceph OSD bootstrap keyring
+        ceph-authtool --create-keyring /var/lib/ceph/bootstrap-osd/ceph.keyring \
+          --gen-key -n client.bootstrap-osd \
+          --cap mon 'profile bootstrap-osd' \
+          --cap mgr 'allow r'
+
+        # Import admin and OSD bootstrap keyrings to mon keyring
+        ceph-authtool /tmp/ceph.mon.keyring --import-keyring /etc/ceph/ceph.client.admin.keyring
+        ceph-authtool /tmp/ceph.mon.keyring --import-keyring /var/lib/ceph/bootstrap-osd/ceph.keyring
+
+        # Create initial monitor map
+        monmaptool --create --add ${HOSTNAME} 127.0.0.1 --fsid ${UUID} /tmp/monmap
+
+        # Setup the monitor daemon
+        ceph-mon --mkfs -i ${HOSTNAME} --monmap /tmp/monmap --keyring /tmp/ceph.mon.keyring
+
+        # Startup ceph-mon daemon - defaults to run in the background
+        ceph-mon --id ${HOSTNAME}
+        ceph mon enable-msgr2
+
+        # ceph-mgr setup
+        mkdir -p /var/lib/ceph/mgr/ceph-${HOSTNAME}
+
+        # Generate cephx key with mgr permissions
+        ceph auth get-or-create mgr.${HOSTNAME} \
+          mon 'allow profile mgr' \
+          osd 'allow *' \
+          mds 'allow *' > /var/lib/ceph/mgr/ceph-${HOSTNAME}/keyring
+
+        # Startup ceph-mgr daemon
+        ceph-mgr --id ${HOSTNAME}
+        ceph health
+
+        # Bootstrap three OSD daemons
+        for osd in `seq 0 2`; do
+          OSD_UUID=$(uuidgen)
+          OSD_SECRET=$(ceph-authtool --gen-print-key)
+          # Generate new OSD ID
+          OSD_ID=$(echo "{\"cephx_secret\": \"$OSD_SECRET\"}" | ceph osd new ${OSD_UUID} -i - -n client.bootstrap-osd -k /var/lib/ceph/bootstrap-osd/ceph.keyring)
+          mkdir -p /var/lib/ceph/osd/ceph-${OSD_ID}
+          # Generate keyring
+          ceph-authtool --create-keyring /var/lib/ceph/osd/ceph-${OSD_ID}/keyring \
+            --name osd.${OSD_ID} --add-key $OSD_SECRET
+          # Create block 'device'
+          dd if=/dev/zero of=/var/lib/ceph/osd/ceph-${OSD_ID}/block bs=4M count=512
+          # Create the OSD
+          ceph-osd --mkfs --id ${OSD_ID} --osd-uuid ${OSD_UUID}
+          # Startup the ceph-osd daemon
+          ceph-osd --id ${OSD_ID}
+        done
+
+        # Check status
+        ceph health | grep -q HEALTH_OK || {
+          cat /var/log/ceph/*.log
+          ceph -s
+          ceph osd tree
+          exit 1
+        }
+
+        ceph osd pool create wolfi-test
+        ceph osd pool application enable wolfi-test rbd
+
+        rbd create --size 20 wolfi-test/small-disk
+        rbd info wolfi-test/small-disk
+        rbd resize --size 50 wolfi-test/small-disk
+        rbd info wolfi-test/small-disk
+        rbd resize --size 30 --allow-shrink wolfi-test/small-disk
+        rbd info wolfi-test/small-disk
+        rbd rm wolfi-test/small-disk
+
+        ceph osd pool autoscale-status
+
+        ceph -s
```

Comment:
Should we `| grep` the output for desired output?

Reply:
This is an output dump on test failure and not the test condition itself.
The complete code snippet is here
```
        ceph health | grep -q HEALTH_OK || {
          cat /var/log/ceph/*.log
          ceph -s
          ceph osd tree
          exit 1
        }
```